src/duckling/game/drawable/animated-drawable.component.ts

```diff
@@ -120,20 +123,28 @@ export class AnimatedDrawableComponent {
         AutoCreateAnimationDialogComponent.open(this._viewContainerRef).subscribe(result => this._autoCreateAnimationFromTilesheet(result));
     }
 
+
     private _autoCreateAnimationFromTilesheet(dialogResult : AutoCreateDialogResult) {
         if (!dialogResult || !dialogResult.imageKey) {
             return;
         }
 
         let frames : ImageDrawable[] = [];
+        let curX = 0;
+        let curY = 0;
+        let textureWidth = this._assets.getImageAssetDimensions({key: dialogResult.imageKey, type: "TexturePNG"}).width;
         for (let i = 0; i < dialogResult.numFrames; i++) {
```

Comment:
This would be clearer as a two dimensional loop
```
for (let curY = 0; frames.length < dialogResult.numFrames; curY += dialogResult.frameDimensions.y) {
     for (let curX = 0; frames.length < dialogResult.numFrames && curX < textureWidth) {

    }
} 
```